File: .sdk-repos/version-8.9/orchestration-cluster-api-js/examples/admin-operations.ts
// Compilable usage examples for cluster variables, system configuration,
// global task listeners, clock, expression, and other admin operations.
// These examples are type-checked during build to guard against API regressions.

import {
  type AuditLogKey,
  createCamundaClient,
  type GlobalListenerId,
  type ProcessDefinitionKey,
  type TenantId,
} from '@camunda8/orchestration-cluster-api';

//#region GetGlobalClusterVariable
async function getGlobalClusterVariableExample() {
  const camunda = createCamundaClient();

  const variable = await camunda.getGlobalClusterVariable(
    { name: 'feature-flags' },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`${variable.name} = ${variable.value}`);
}
//#endregion GetGlobalClusterVariable

//#region CreateGlobalClusterVariable
async function createGlobalClusterVariableExample() {
  const camunda = createCamundaClient();

  const result = await camunda.createGlobalClusterVariable({
    name: 'feature-flags',
    value: { darkMode: true },
  });

  console.log(`Created: ${result.name}`);
}
//#endregion CreateGlobalClusterVariable

//#region UpdateGlobalClusterVariable
async function updateGlobalClusterVariableExample() {
  const camunda = createCamundaClient();

  await camunda.updateGlobalClusterVariable({
    name: 'feature-flags',
    value: { darkMode: false },
  });
}
//#endregion UpdateGlobalClusterVariable

//#region DeleteGlobalClusterVariable
async function deleteGlobalClusterVariableExample() {
  const camunda = createCamundaClient();

  await camunda.deleteGlobalClusterVariable({ name: 'feature-flags' });
}
//#endregion DeleteGlobalClusterVariable

//#region GetTenantClusterVariable
async function getTenantClusterVariableExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const variable = await camunda.getTenantClusterVariable(
    {
      tenantId,
      name: 'config',
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`${variable.name} = ${variable.value}`);
}
//#endregion GetTenantClusterVariable

//#region CreateTenantClusterVariable
async function createTenantClusterVariableExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.createTenantClusterVariable({
    tenantId,
    name: 'config',
    value: { region: 'us-east-1' },
  });

  console.log(`Created: ${result.name}`);
}
//#endregion CreateTenantClusterVariable

//#region UpdateTenantClusterVariable
async function updateTenantClusterVariableExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  await camunda.updateTenantClusterVariable({
    tenantId,
    name: 'config',
    value: { region: 'eu-west-1' },
  });
}
//#endregion UpdateTenantClusterVariable

//#region DeleteTenantClusterVariable
async function deleteTenantClusterVariableExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  await camunda.deleteTenantClusterVariable({
    tenantId,
    name: 'config',
  });
}
//#endregion DeleteTenantClusterVariable

//#region SearchClusterVariables
async function searchClusterVariablesExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchClusterVariables(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const variable of result.items ?? []) {
    console.log(`${variable.name} = ${variable.value}`);
  }
}
//#endregion SearchClusterVariables

//#region CreateGlobalTaskListener
async function createGlobalTaskListenerExample(id: GlobalListenerId) {
  const camunda = createCamundaClient();

  const result = await camunda.createGlobalTaskListener({
    id,
    eventTypes: ['completing'],
    type: 'audit-log-listener',
  });

  console.log(`Created listener: ${result.id}`);
}
//#endregion CreateGlobalTaskListener

//#region GetGlobalTaskListener
async function getGlobalTaskListenerExample(id: GlobalListenerId) {
  const camunda = createCamundaClient();

  const listener = await camunda.getGlobalTaskListener(
    { id },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Listener: ${listener.type} (${listener.eventTypes})`);
}
//#endregion GetGlobalTaskListener

//#region UpdateGlobalTaskListener
async function updateGlobalTaskListenerExample(id: GlobalListenerId) {
  const camunda = createCamundaClient();

  await camunda.updateGlobalTaskListener({
    id,
    eventTypes: ['completing'],
    type: 'updated-audit-listener',
  });
}
//#endregion UpdateGlobalTaskListener

//#region DeleteGlobalTaskListener
async function deleteGlobalTaskListenerExample(id: GlobalListenerId) {
  const camunda = createCamundaClient();

  await camunda.deleteGlobalTaskListener({
    id,
  });
}
//#endregion DeleteGlobalTaskListener

//#region SearchGlobalTaskListeners
async function searchGlobalTaskListenersExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchGlobalTaskListeners(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const listener of result.items ?? []) {
    console.log(`${listener.id}: ${listener.type} (${listener.eventTypes})`);
  }
}
//#endregion SearchGlobalTaskListeners

//#region GetLicense
async function getLicenseExample() {
  const camunda = createCamundaClient();

  const license = await camunda.getLicense();

  console.log(`License type: ${license.validLicense}`);
}
//#endregion GetLicense

//#region GetSystemConfiguration
async function getSystemConfigurationExample() {
  const camunda = createCamundaClient();

  const config = await camunda.getSystemConfiguration();

  console.log(`Configuration loaded: ${JSON.stringify(config)}`);
}
//#endregion GetSystemConfiguration

//#region GetAuthentication
async function getAuthenticationExample() {
  const camunda = createCamundaClient();

  const user = await camunda.getAuthentication();

  console.log(`Authenticated as: ${user.username}`);
}
//#endregion GetAuthentication

//#region GetStatus
async function getStatusExample() {
  const camunda = createCamundaClient();

  await camunda.getStatus();

  console.log('Cluster is healthy');
}
//#endregion GetStatus

//#region PinClock
async function pinClockExample() {
  const camunda = createCamundaClient();

  await camunda.pinClock({
    timestamp: 1735689599000,
  });

  console.log('Clock pinned');
}
//#endregion PinClock

//#region ResetClock
async function resetClockExample() {
  const camunda = createCamundaClient();

  await camunda.resetClock();

  console.log('Clock reset');
}
//#endregion ResetClock

//#region EvaluateConditionals
async function evaluateConditionalsExample(tenantId: TenantId) {
  const camunda = createCamundaClient();

  const result = await camunda.evaluateConditionals({
    variables: { orderReady: true },
    tenantId,
  });

  console.log(`Evaluated conditionals: ${JSON.stringify(result)}`);
}
//#endregion EvaluateConditionals

//#region EvaluateExpression
async function evaluateExpressionExample() {
  const camunda = createCamundaClient();

  const result = await camunda.evaluateExpression({
    expression: '= x + y',
    variables: { x: 10, y: 20 },
  });

  console.log(`Result: ${result.result}`);
}
//#endregion EvaluateExpression

//#region GetResource
async function getResourceExample(resourceKey: ProcessDefinitionKey) {
  const camunda = createCamundaClient();

  const resource = await camunda.getResource({
    resourceKey,
  });

  console.log(`Resource: ${resource.resourceName} (${resource.resourceId})`);
}
//#endregion GetResource

//#region GetResourceContent
async function getResourceContentExample(resourceKey: ProcessDefinitionKey) {
  const camunda = createCamundaClient();

  const content = await camunda.getResourceContent({
    resourceKey,
  });

  console.log(`Content retrieved (type: ${typeof content})`);
}
//#endregion GetResourceContent

//#region GetUsageMetrics
async function getUsageMetricsExample() {
  const camunda = createCamundaClient();

  const metrics = await camunda.getUsageMetrics(
    {
      startTime: '2025-01-01T00:00:00Z',
      endTime: '2025-12-31T23:59:59Z',
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Usage metrics retrieved: ${JSON.stringify(metrics)}`);
}
//#endregion GetUsageMetrics

//#region SearchMessageSubscriptions
async function searchMessageSubscriptionsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchMessageSubscriptions(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const sub of result.items ?? []) {
    console.log(`Subscription: ${sub.messageName}`);
  }
}
//#endregion SearchMessageSubscriptions

//#region SearchCorrelatedMessageSubscriptions
async function searchCorrelatedMessageSubscriptionsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchCorrelatedMessageSubscriptions(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const sub of result.items ?? []) {
    console.log(`Correlated subscription: ${sub.messageName}`);
  }
}
//#endregion SearchCorrelatedMessageSubscriptions

//#region GetAuditLog
async function getAuditLogExample(auditLogKey: AuditLogKey) {
  const camunda = createCamundaClient();

  const log = await camunda.getAuditLog({ auditLogKey }, { consistency: { waitUpToMs: 5000 } });

  console.log(`Audit log: ${log.operationType}`);
}
//#endregion GetAuditLog

//#region SearchAuditLogs
async function searchAuditLogsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchAuditLogs(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const log of result.items ?? []) {
    console.log(`${log.auditLogKey}: ${log.operationType}`);
  }
}
//#endregion SearchAuditLogs

//#region GetProcessInstanceStatisticsByError
async function getProcessInstanceStatisticsByErrorExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getProcessInstanceStatisticsByError(
    {},
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const stat of result.items ?? []) {
    console.log(`Error: ${stat.errorMessage}, count: ${stat.activeInstancesWithErrorCount}`);
  }
}
//#endregion GetProcessInstanceStatisticsByError

//#region GetProcessInstanceStatisticsByDefinition
async function getProcessInstanceStatisticsByDefinitionExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getProcessInstanceStatisticsByDefinition(
    {
      filter: {
        errorHashCode: 12345,
      },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const stat of result.items ?? []) {
    console.log(
      `Definition ${stat.processDefinitionId}: ${stat.activeInstancesWithErrorCount} incidents`
    );
  }
}
//#endregion GetProcessInstanceStatisticsByDefinition

//#region GetJobErrorStatistics
async function getJobErrorStatisticsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getJobErrorStatistics(
    {
      filter: {
        from: '2025-01-01T00:00:00Z',
        to: '2025-12-31T23:59:59Z',
        jobType: 'payment-processing',
      },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const stat of result.items ?? []) {
    console.log(`Error: ${stat.errorMessage}, workers: ${stat.workers}`);
  }
}
//#endregion GetJobErrorStatistics

//#region GetJobTimeSeriesStatistics
async function getJobTimeSeriesStatisticsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getJobTimeSeriesStatistics(
    {
      filter: {
        from: '2025-01-01T00:00:00Z',
        to: '2025-12-31T23:59:59Z',
        jobType: 'payment-processing',
      },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const point of result.items ?? []) {
    console.log(`Time: ${point.time}, created: ${point.created.count}`);
  }
}
//#endregion GetJobTimeSeriesStatistics

//#region GetJobTypeStatistics
async function getJobTypeStatisticsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getJobTypeStatistics({}, { consistency: { waitUpToMs: 5000 } });

  for (const stat of result.items ?? []) {
    console.log(`Type: ${stat.jobType}, workers: ${stat.workers}`);
  }
}
//#endregion GetJobTypeStatistics

//#region GetJobWorkerStatistics
async function getJobWorkerStatisticsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getJobWorkerStatistics(
    {
      filter: {
        from: '2025-01-01T00:00:00Z',
        to: '2025-12-31T23:59:59Z',
        jobType: 'payment-processing',
      },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const stat of result.items ?? []) {
    console.log(`Worker: ${stat.worker}, completed: ${stat.completed.count}`);
  }
}
//#endregion GetJobWorkerStatistics

//#region GetGlobalJobStatistics
async function getGlobalJobStatisticsExample() {
  const camunda = createCamundaClient();

  const result = await camunda.getGlobalJobStatistics(
    {
      from: '2025-01-01T00:00:00Z',
      to: '2025-12-31T23:59:59Z',
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Statistics retrieved: ${JSON.stringify(result)}`);
}
//#endregion GetGlobalJobStatistics

// Suppress "declared but never read"
void getGlobalClusterVariableExample;
void createGlobalClusterVariableExample;
void updateGlobalClusterVariableExample;
void deleteGlobalClusterVariableExample;
void getTenantClusterVariableExample;
void createTenantClusterVariableExample;
void updateTenantClusterVariableExample;
void deleteTenantClusterVariableExample;
void searchClusterVariablesExample;
void createGlobalTaskListenerExample;
void getGlobalTaskListenerExample;
void updateGlobalTaskListenerExample;
void deleteGlobalTaskListenerExample;
void searchGlobalTaskListenersExample;
void getLicenseExample;
void getSystemConfigurationExample;
void getAuthenticationExample;
void getStatusExample;
void pinClockExample;
void resetClockExample;
void evaluateConditionalsExample;
void evaluateExpressionExample;
void getResourceExample;
void getResourceContentExample;
void getUsageMetricsExample;
void searchMessageSubscriptionsExample;
void searchCorrelatedMessageSubscriptionsExample;
void getAuditLogExample;
void searchAuditLogsExample;
void getProcessInstanceStatisticsByErrorExample;
void getProcessInstanceStatisticsByDefinitionExample;
void getJobErrorStatisticsExample;
void getJobTimeSeriesStatisticsExample;
void getJobTypeStatisticsExample;
void getJobWorkerStatisticsExample;
void getGlobalJobStatisticsExample;

File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/admin_operations.py
# Compilable usage examples for admin, system, and statistics operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

import datetime

from camunda_orchestration_sdk import (
    AuditLogKey,
    AuditLogSearchQueryRequest,
    CamundaClient,
    ClockPinRequest,
    ClusterVariableSearchQueryRequest,
    ConditionalEvaluationInstruction,
    ConditionalEvaluationInstructionVariables,
    CorrelatedMessageSubscriptionSearchQuery,
    CreateClusterVariableRequest,
    CreateClusterVariableRequestValue,
    CreateGlobalTaskListenerRequest,
    ExpressionEvaluationRequest,
    GlobalListenerId,
    GlobalTaskListenerEventTypeEnum,
    GlobalTaskListenerSearchQueryRequest,
    IncidentProcessInstanceStatisticsByDefinitionQuery,
    IncidentProcessInstanceStatisticsByDefinitionQueryFilter,
    IncidentProcessInstanceStatisticsByErrorQuery,
    JobErrorStatisticsFilter,
    JobErrorStatisticsQuery,
    JobTimeSeriesStatisticsFilter,
    JobTimeSeriesStatisticsQuery,
    JobTypeStatisticsQuery,
    JobWorkerStatisticsFilter,
    JobWorkerStatisticsQuery,
    MessageSubscriptionSearchQuery,
    TenantId,
    Unset,
    UpdateClusterVariableRequest,
    UpdateClusterVariableRequestValue,
    UpdateGlobalTaskListenerRequest,
)


# region GetGlobalClusterVariable
def get_global_cluster_variable_example() -> None:
    client = CamundaClient()

    result = client.get_global_cluster_variable(name="my-variable")

    print(f"Variable: {result.name} = {result.value}")
# endregion GetGlobalClusterVariable


# region CreateGlobalClusterVariable
def create_global_cluster_variable_example() -> None:
    client = CamundaClient()

    result = client.create_global_cluster_variable(
        data=CreateClusterVariableRequest(
            name="my-variable",
            value=CreateClusterVariableRequestValue.from_dict({"key": "my-value"}),
        ),
    )

    print(f"Created variable: {result.name}")
# endregion CreateGlobalClusterVariable


# region UpdateGlobalClusterVariable
def update_global_cluster_variable_example() -> None:
    client = CamundaClient()

    result = client.update_global_cluster_variable(
        name="my-variable",
        data=UpdateClusterVariableRequest(
            value=UpdateClusterVariableRequestValue.from_dict({"key": "updated-value"}),
        ),
    )

    print(f"Updated variable: {result.name}")
# endregion UpdateGlobalClusterVariable


# region DeleteGlobalClusterVariable
def delete_global_cluster_variable_example() -> None:
    client = CamundaClient()

    client.delete_global_cluster_variable(name="my-variable")
# endregion DeleteGlobalClusterVariable


# region GetTenantClusterVariable
def get_tenant_cluster_variable_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.get_tenant_cluster_variable(
        tenant_id=tenant_id,
        name="my-variable",
    )

    print(f"Variable: {result.name} = {result.value}")
# endregion GetTenantClusterVariable


# region CreateTenantClusterVariable
def create_tenant_cluster_variable_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.create_tenant_cluster_variable(
        tenant_id=tenant_id,
        data=CreateClusterVariableRequest(
            name="my-variable",
            value=CreateClusterVariableRequestValue.from_dict({"key": "tenant-value"}),
        ),
    )

    print(f"Created variable: {result.name}")
# endregion CreateTenantClusterVariable


# region UpdateTenantClusterVariable
def update_tenant_cluster_variable_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.update_tenant_cluster_variable(
        tenant_id=tenant_id,
        name="my-variable",
        data=UpdateClusterVariableRequest(
            value=UpdateClusterVariableRequestValue.from_dict({"key": "updated-tenant-value"}),
        ),
    )

    print(f"Updated variable: {result.name}")
# endregion UpdateTenantClusterVariable


# region DeleteTenantClusterVariable
def delete_tenant_cluster_variable_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.delete_tenant_cluster_variable(
        tenant_id=tenant_id,
        name="my-variable",
    )
# endregion DeleteTenantClusterVariable


# region SearchClusterVariables
def search_cluster_variables_example() -> None:
    client = CamundaClient()

    result = client.search_cluster_variables(
        data=ClusterVariableSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for var in result.items:
            print(f"Variable: {var.name}")
# endregion SearchClusterVariables


# region CreateGlobalTaskListener
def create_global_task_listener_example() -> None:
    client = CamundaClient()

    result = client.create_global_task_listener(
        data=CreateGlobalTaskListenerRequest(
            id="audit-log-listener",
            event_types=[GlobalTaskListenerEventTypeEnum.COMPLETING],
            type_="my-task-listener",
        ),
    )

    print(f"Task listener: {result.id}")
# endregion CreateGlobalTaskListener


# region GetGlobalTaskListener
def get_global_task_listener_example(listener_id: GlobalListenerId) -> None:
    client = CamundaClient()

    result = client.get_global_task_listener(id=listener_id)

    print(f"Task listener: {result.event_types}")
# endregion GetGlobalTaskListener


# region UpdateGlobalTaskListener
def update_global_task_listener_example(listener_id: GlobalListenerId) -> None:
    client = CamundaClient()

    result = client.update_global_task_listener(
        id=listener_id,
        data=UpdateGlobalTaskListenerRequest(
            event_types=[GlobalTaskListenerEventTypeEnum.COMPLETING],
            type_="updated-task-listener",
        ),
    )

    print(f"Updated listener: {result.id}")
# endregion UpdateGlobalTaskListener


# region DeleteGlobalTaskListener
def delete_global_task_listener_example(listener_id: GlobalListenerId) -> None:
    client = CamundaClient()

    client.delete_global_task_listener(id=listener_id)
# endregion DeleteGlobalTaskListener


# region SearchGlobalTaskListeners
def search_global_task_listeners_example() -> None:
    client = CamundaClient()

    result = client.search_global_task_listeners(
        data=GlobalTaskListenerSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for listener in result.items:
            print(f"Listener: {listener.id}")
# endregion SearchGlobalTaskListeners


# region GetLicense
def get_license_example() -> None:
    client = CamundaClient()

    result = client.get_license()

    print(f"License type: {result.license_type}")
# endregion GetLicense


# region GetSystemConfiguration
def get_system_configuration_example() -> None:
    client = CamundaClient()

    result = client.get_system_configuration()

    print(f"System config: {result}")
# endregion GetSystemConfiguration


# region GetAuthentication
def get_authentication_example() -> None:
    client = CamundaClient()

    result = client.get_authentication()

    print(f"Authenticated user: {result.username}")
# endregion GetAuthentication


# region GetTopology
def get_topology_example() -> None:
    client = CamundaClient()

    result = client.get_topology()

    print(f"Topology: {result}")
# endregion GetTopology


# region GetStatus
def get_status_example() -> None:
    client = CamundaClient()

    client.get_status()

    print("Cluster is healthy")
# endregion GetStatus


# region PinClock
def pin_clock_example() -> None:
    client = CamundaClient()

    client.pin_clock(
        data=ClockPinRequest(
            timestamp=1700000000000,
        ),
    )
# endregion PinClock


# region ResetClock
def reset_clock_example() -> None:
    client = CamundaClient()

    client.reset_clock()
# endregion ResetClock


# region EvaluateConditionals
def evaluate_conditionals_example() -> None:
    client = CamundaClient()

    result = client.evaluate_conditionals(
        data=ConditionalEvaluationInstruction(
            variables=ConditionalEvaluationInstructionVariables.from_dict({"orderReady": True}),
        ),
    )

    print(f"Result: {result}")
# endregion EvaluateConditionals


# region EvaluateExpression
def evaluate_expression_example() -> None:
    client = CamundaClient()

    result = client.evaluate_expression(
        data=ExpressionEvaluationRequest(
            expression="= 1 + 2",
        ),
    )

    print(f"Result: {result.result}")
# endregion EvaluateExpression


# region GetResource
def get_resource_example() -> None:
    client = CamundaClient()

    result = client.get_resource(resource_key="123456")

    print(f"Resource: {result.resource_name}")
# endregion GetResource


# region GetResourceContent
def get_resource_content_example() -> None:
    client = CamundaClient()

    content = client.get_resource_content(resource_key="123456")

    print(f"Content length: {len(content)}")
# endregion GetResourceContent


# region GetUsageMetrics
def get_usage_metrics_example() -> None:
    client = CamundaClient()

    result = client.get_usage_metrics(
        start_time=datetime.datetime(2024, 1, 1),
        end_time=datetime.datetime(2024, 12, 31),
    )

    print(f"Metrics: {result}")
# endregion GetUsageMetrics


# region SearchMessageSubscriptions
def search_message_subscriptions_example() -> None:
    client = CamundaClient()

    result = client.search_message_subscriptions(
        data=MessageSubscriptionSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for sub in result.items:
            print(f"Subscription: {sub.message_name}")
# endregion SearchMessageSubscriptions


# region SearchCorrelatedMessageSubscriptions
def search_correlated_message_subscriptions_example() -> None:
    client = CamundaClient()

    result = client.search_correlated_message_subscriptions(
        data=CorrelatedMessageSubscriptionSearchQuery(),
    )

    if not isinstance(result.items, Unset):
        for sub in result.items:
            print(f"Correlated subscription: {sub.message_name}")
# endregion SearchCorrelatedMessageSubscriptions


# region GetAuditLog
def get_audit_log_example(audit_log_key: AuditLogKey) -> None:
    client = CamundaClient()

    result = client.get_audit_log(audit_log_key=audit_log_key)

    print(f"Audit log: {result.audit_log_key}")
# endregion GetAuditLog


# region SearchAuditLogs
def search_audit_logs_example() -> None:
    client = CamundaClient()

    result = client.search_audit_logs(
        data=AuditLogSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for log in result.items:
            print(f"Audit log: {log.audit_log_key}")
# endregion SearchAuditLogs


# region GetProcessInstanceStatisticsByError
def get_process_instance_statistics_by_error_example() -> None:
    client = CamundaClient()

    result = client.get_process_instance_statistics_by_error(
        data=IncidentProcessInstanceStatisticsByErrorQuery(),
    )

    if not isinstance(result.items, Unset):
        for stat in result.items:
            print(f"Error: {stat.error_message}")
# endregion GetProcessInstanceStatisticsByError


# region GetProcessInstanceStatisticsByDefinition
def get_process_instance_statistics_by_definition_example() -> None:
    client = CamundaClient()

    result = client.get_process_instance_statistics_by_definition(
        data=IncidentProcessInstanceStatisticsByDefinitionQuery(
            filter_=IncidentProcessInstanceStatisticsByDefinitionQueryFilter(
                error_hash_code=12345,
            ),
        ),
    )

    if not isinstance(result.items, Unset):
        for stat in result.items:
            print(f"Definition: {stat.process_definition_key}")
# endregion GetProcessInstanceStatisticsByDefinition


# region GetJobErrorStatistics
def get_job_error_statistics_example() -> None:
    client = CamundaClient()

    result = client.get_job_error_statistics(
        data=JobErrorStatisticsQuery(
            filter_=JobErrorStatisticsFilter(
                from_=datetime.datetime(2024, 1, 1),
                to=datetime.datetime(2024, 12, 31),
                job_type="payment-processing",
            ),
        ),
    )

    if not isinstance(result.items, Unset):
        for stat in result.items:
            print(f"Error: {stat.error_code}")
# endregion GetJobErrorStatistics


# region GetJobTimeSeriesStatistics
def get_job_time_series_statistics_example() -> None:
    client = CamundaClient()

    result = client.get_job_time_series_statistics(
        data=JobTimeSeriesStatisticsQuery(
            filter_=JobTimeSeriesStatisticsFilter(
                from_=datetime.datetime(2024, 1, 1),
                to=datetime.datetime(2024, 12, 31),
                job_type="payment-processing",
            ),
        ),
    )

    if not isinstance(result.items, Unset):
        for stat in result.items:
            print(f"Time series: {stat}")
# endregion GetJobTimeSeriesStatistics


# region GetJobTypeStatistics
def get_job_type_statistics_example() -> None:
    client = CamundaClient()

    result = client.get_job_type_statistics(
        data=JobTypeStatisticsQuery(),
    )

    if not isinstance(result.items, Unset):
        for stat in result.items:
            print(f"Job type: {stat.job_type}")
# endregion GetJobTypeStatistics


# region GetJobWorkerStatistics
def get_job_worker_statistics_example() -> None:
    client = CamundaClient()

    result = client.get_job_worker_statistics(
        data=JobWorkerStatisticsQuery(
            filter_=JobWorkerStatisticsFilter(
                from_=datetime.datetime(2024, 1, 1),
                to=datetime.datetime(2024, 12, 31),
                job_type="payment-processing",
            ),
        ),
    )

    if not isinstance(result.items, Unset):
        for stat in result.items:
            print(f"Worker: {stat.worker}")
# endregion GetJobWorkerStatistics


# region GetGlobalJobStatistics
def get_global_job_statistics_example() -> None:
    client = CamundaClient()

    result = client.get_global_job_statistics(
        from_=datetime.datetime(2024, 1, 1),
        to=datetime.datetime(2024, 12, 31),
    )

    print(f"Global job stats: {result}")
# endregion GetGlobalJobStatistics
